spec/factories/applicants.rb

```diff
@@ -10,6 +10,10 @@
       addresses { build_list :address, 1 }
     end
 
+    trait :with_address_for_xml_fixture do
```

Comment:
This doesn't seem related to the rest of the PR?

Reply:
it is actually - it resolves a flickering test which other changes here made worse - I couldn't work out why. But this ensures that the applicant is created with the correct postcode for the xml fixture file.